README.md

```diff
@@ -90,25 +93,23 @@ let info fmt = Irmin_unix.info ~author fmt
 
 let main =
   (* Open the repo *)
-  Store.Repo.v config >>=
-
+  Store.Repo.v config >>= fun repo ->
   (* Load the master branch *)
-  Store.master >>= fun t ->
-
+  Store.master repo >>= fun t ->
   (* Set key "foo/bar" to "testing 123" *)
-  Store.set_exn t ~info:(info "Updating foo/bar") ["foo"; "bar"] "testing 123" >>= fun () ->
-
-  (* Get key "foo/bar" and print it to stdout *)
-  Store.get t ["foo"; "bar"] >|= fun x ->
-  Printf.printf "foo/bar => '%s'\n" x
+  Store.set_exn t ~info:(info "Updating foo/bar") [ "foo"; "bar" ] "testing 123"
+  >>= fun () ->
+  (* Get key "foo/bar" print it to stdout *)
+  Store.get t [ "foo"; "bar" ] >|= fun x -> Printf.printf "foo/bar => '%s'\n" x
 
 (* Run the program *)
 let () = Lwt_main.run main
 ```
 
 The example is contained in `examples/readme.ml`. It can be compiled and executed with dune:
 
-```bash
+<!-- $MDX skip -->
```

Comment:
Why is this skipped?

Reply:
I think `mdx` has issues with running Dune from inside Dune:

```diff
 |```sh
 |$ dune build examples/readme.exe
-|$ dune exec examples/readme.exe
-|foo/bar => 'testing 123'
+|Error: Multiple rules generated for _build/default/examples/readme.exe:
+|- file present in source tree
+|- examples/dune:2
+|Hint: rm -f examples/readme.exe
+|[1]
+|$ dune exec examples/readme.exe
+|Error: Multiple rules generated for _build/default/examples/readme.exe:
+|- file present in source tree
+|- examples/dune:2
+|Hint: rm -f examples/readme.exe
+|[1]
```

but I may just be missing something :slightly_smiling_face:

Reply:
that's weird, as it's something that is done in RWO without much trouble. Again please report it upstream so that it's tracked and solved properly :-)

Reply:
Looks to me like RWO is cheating slightly by unsetting `INSIDE_DUNE`: https://github.com/realworldocaml/book/blame/master/book/testing/README.md#L544. I'll discuss this with @NathanReb when he gets back from holiday next week.

Reply:
Please open an issue so I can follow the discussion too ;-)